### PR TITLE
feat: add support and example modules for hibernate jakarta & javax version

### LINKED ISSUE #441: Support hibernate reactive
### Requirements
- Please implement the hibernate-reactive-support module to help execute the JPQL query using only `Hibernate-Reactive`.
- You should implement two modules for javax and jakarta.
- You should implement unit tests and an integration test, and the integration test is implemented through the example module.

### PR DESCRIPTION
**Pull request by saeyoon94**
# Motivation:

* Implements hibernate-support module and hibernate-example module for both jakarta and javax version

# Modifications:

* List the modifications you've made in detail.

# Commit Convention Rule

* Please commit your modification based by [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/)
* This commit convention is referred from [angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines)

| Commit type | Description                                             |
|-------------|---------------------------------------------------------|
| feat        | New Feature                                             |
| fix         | Fix bug                                                 |
| docs        | Documentation only changed                              |
| ci          | Change CI configuration                                 |
| refactor    | Not a bug fix or add feature, just refactoring code     |
| test        | Add Test case or fix wrong test case                    |
| style       | Only change the code style(ex. white-space, formatting) |
| chore       | It refers to minor tasks such as library version upgrade, typo correction, etc. |

* If you want to add some more `commit type` please describe it on the **Pull Request**

# Result:

Closes #441

### CODE REVIEW

**Comment by cj848** [comment]
All the code in the support module appears to be completely identical to spring-data-jpa.
How about creating a persistence-api(-javax)-support (tentative name) module and importing it into the hibernate and spring-data-jpa modules?
I don't think it would be a bad idea to have dependencies within support.
@shouwn Please give me your opinion.

-- korean
support 모듈 내의 모든 코드가 spring-data-jpa 와 완전 동일한걸로 보이는데..
persistence-api(-javax)-support (가칭) 모듈을 만들고 그걸 hibernate, spring-data-jpa 모듈에 같이 import 하면 어떨까요?
support 안에서의 의존성 가져가는건 나쁘지 않을 것 같습니다. 
@shouwn 의견 부탁드립니다.

**Comment by shouwn** [comment]
I've thought about what you said. In 3.0.0, I will allow duplicates and consider these later.

For `DSL` and `Render`, I have some insight because I has been working on them since 2.X, but I'm not sure what the `support` will look like. `support` is something I'm new to and developing now, so we're not sure what it will look like.

If we create a common set of support like jpa-support-core, 
you're going to have dependency complexity, and how far to isolate the logic and how to divide the interfaces is a really hard topic.

For now, I'm going to allow all the duplication, and when it gets harder for us to maintain and we get some solid insights from looking at the SUPPORT modules, then we'll create it.

It's a matter of how we're going to maintain our internal code anyway, so it doesn't affect our external users, so that's how we decided to do it.

**Comment by cj848** [comment]
@saeyoon94 please check failed test.

File: example/hibernate-javax/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    annotation("javax.persistence.Entity")
+    annotation("javax.persistence.Embeddable")
+}
+
+allOpen {
+    annotation("com.linecorp.kotlinjdsl.example.jpql.hibernate.javax.annotation.CompositeId")
+    annotation("javax.persistence.Entity")
+    annotation("javax.persistence.Embeddable")
+}
+
```
Comment (cj848) [inline example/hibernate-javax/build.gradle.kts:34]:
```suggestion
kover {
     excludeInstrumentation {
         packages("org.hibernate.*")
     }
}
```
Adding this line seems to resolve the test report failure.
When kover performs a bytecode instrument to measure coverage, it conflicts with the bytecode instrument performed by hibernate.

File: support/hibernate-javax/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    compileOnly(projects.jpqlQueryModel)
+    compileOnly(projects.jpqlRender)
+
+    testImplementation(libs.spring.boot2.starter)
+    testImplementation(libs.javax.hibernate.core)
+    testImplementation(projects.jpqlDsl)
+    testImplementation(projects.jpqlQueryModel)
+    testImplementation(projects.jpqlRender)
+}
+
```
Comment (cj848) [inline support/hibernate-javax/build.gradle.kts:16]:
```suggestion
kover {
     excludeInstrumentation {
         packages("org.hibernate.*")
     }
}
```
Adding this line seems to resolve the test report failure.
When kover performs a bytecode instrument to measure coverage, it conflicts with the bytecode instrument performed by hibernate.

File: support/hibernate/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    compileOnly(projects.jpqlQueryModel)
+    compileOnly(projects.jpqlRender)
+
+    testImplementation(libs.spring.boot3.starter)
+    testImplementation(libs.jakarta.hibernate.core)
+    testImplementation(projects.jpqlDsl)
+    testImplementation(projects.jpqlQueryModel)
+    testImplementation(projects.jpqlRender)
+}
+
```
Comment (cj848) [inline support/hibernate/build.gradle.kts:16]:
```suggestion
kover {
     excludeInstrumentation {
         packages("org.hibernate.*")
     }
}
```
Adding this line seems to resolve the test report failure.
When kover performs a bytecode instrument to measure coverage, it conflicts with the bytecode instrument performed by hibernate.

File: support/hibernate-javax/build.gradle.kts (hunk 2 above)
Comment (saeyoon94) [reply support/hibernate-javax/build.gradle.kts:14]: I have applied the review.

File: support/hibernate/build.gradle.kts (hunk 3 above)
Comment (saeyoon94) [reply support/hibernate/build.gradle.kts:14]: I have applied the review.

**Comment by cj848** [comment]
check ktlint please.

**Pull request merged**